src/fourier.jl

```diff
@@ -0,0 +1,781 @@
+
+normalize_leahy_from_variance(unnorm_power:: AbstractVector{<:Number}, 
+                              variance::Real, n_bin::Integer) = 
+                                unnorm_power * 2.0 / (variance * n_bin)
+
+normalize_leahy_poisson(unnorm_power:: AbstractVector{<:Number}, 
+                        n_ph::Real) = unnorm_power * 2.0 / n_ph
+
+function normalize_periodograms(unnorm_power:: AbstractVector{<:Number}, dt::Real, 
+                                n_bin::Integer; mean_flux=nothing, n_ph=nothing,
+                                variance=nothing, background_flux::Real=0.0, 
+                                norm::String="frac",power_type::String="all")
+    
+    if norm == "leahy" && !isnothing(variance)
+        pds = normalize_leahy_from_variance(unnorm_power, variance, n_bin)
+    elseif norm == "leahy"
+        pds = normalize_leahy_poisson(unnorm_power, n_ph)
+    elseif norm == "frac"
+        pds = normalize_frac(
+            unnorm_power, dt, n_bin, mean_flux,
+            background_flux=background_flux)
+    elseif norm == "abs"
+        pds = normalize_abs(unnorm_power, dt, n_bin)
+    elseif norm == "none"
+        pds = unnorm_power
+    else 
+        throw(ArgumentError("Incorrect Arguments"))
+    end
+
+    if power_type == "all"
+        return pds
+    elseif power_type == "real"
+        return real(pds)
+    elseif power_type in ["abs", "absolute"]
+        return abs.(pds)
+    else 
+        throw(ArgumentError("Incorrect Arguments"))
+    end
+
+end
+
+function bias_term(power1::T, power2::T, power1_noise::Real, 
+                   power2_noise::Real, n_ave::Integer;
+                   intrinsic_coherence::Real=1.0) where {T<:Union{AbstractVector{<:Real},Real}}
+    
+    if n_ave > 500
+        return 0.0 * power1
+    end
+    return @. power1 * power2 - intrinsic_coherence * (power1 - power1_noise) * (power2 - power2_noise) / n_ave
+end
+
+function _raw_coherence_single(cross_power::Number, power1::Real, power2::Real, 
+                               power1_noise::Real, power2_noise::Real, 
+                               n_ave::Integer; intrinsic_coherence::Real=1.0)
+
+    bsq = bias_term(power1, power2, power1_noise, power2_noise, n_ave;
+                    intrinsic_coherence=intrinsic_coherence)
+    num = real(cross_power .* conj(cross_power)) - bsq
+    if num < 0
+        num = real(cross_power * conj(cross_power))
+    end
+    den = power1 * power2
+    return num / den
+end
+
+raw_coherence(cross_power::T1, power1::T2, power2::T2, power1_noise::Real, 
+              power2_noise::Real, n_ave::Integer;
+              intrinsic_coherence::Real=1.0) where 
+              {T1<:Union{AbstractVector{<:Number},Number},T2<:Union{AbstractVector{<:Real},Real}} = 
+                    _raw_coherence_single.(cross_power, power1, power2, 
+                                       power1_noise, power2_noise, n_ave;
+                                       intrinsic_coherence)
```

Comment:
Again, it sounds like this should be defined on scalars, and use broadcasting on the call site